### DISCUSSION TITLE
How to secure POST, PUT and DELETE but not GET?

### DISCUSSION THREAD

**Discussion by moedeveloper** [ThreeMammals/Ocelot#1988] [Q&A] [answered]
I really would like to know if I am doing it correctly. 

I want to secure POST, PUT and DELETE but not GET, I would like to leave GET with no need for authentication. Is that how to do it? also, how can I avoid the reputation of the repeated attributes? Do I need to have GlobalConfiguration and how to do it?

![Screenshot 2024-03-04 at 12 31 35](https://github.com/ThreeMammals/Ocelot/assets/16238332/adb2fbd7-3f84-4cb9-a731-7d00b4c8ffc6)

### Update by Maintainer 
Please remove this large picture 👆 and insert JSON text!

**Comment by raman-m** [comment, marked as answer]
> I want to secure POST, PUT and DELETE but not GET, I would like to leave GET with no need for authentication. Is that how to do it?

Just don't define the verb!
Or, define 2 different routes with different upstream methods, but upstream templates can be the same.
1st route is authenticated, and the 2nd one is anonymous. 

Note, your UpstreamPathTemplate of the second route includes the template of the first one. 
`/notification/{everything}` contains `/notification/device`

Final config depends on your services. I can't recommend on the config more. Please finish your setup by your own!

-----
> also, how can I avoid the reputation of the repeated attributes?

Sorry? I don't understand you. There is no ability to join routes or define route families... You must define exactly the same props of [route configuration](https://ocelot.readthedocs.io/en/latest/features/configuration.html)!

-----
> Do I need to have GlobalConfiguration and how to do it?

How to do it, what? To define classic routes you needn't use global configuration at all.
Global config has limited number of features! A few of them are: [Dynamic Routing](https://ocelot.readthedocs.io/en/latest/features/routing.html#dynamic-routing), global Rate Liming of **DynamicRoutes** etc.
Read [the docs](https://ocelot.readthedocs.io/en/latest/index.html) carefully!

**Reply by moedeveloper** [reply]
Thank you i got you

**Reply by moedeveloper** [reply]
I have one more question please, is there is any plans to officially add support for Swagger? I know there is this package but just wondering?

https://github.com/Burgyn/MMLib.SwaggerForOcelot

**Reply by raman-m** [reply]
Yes, there are plans!
- #1429 
- [Releases](https://github.com/ThreeMammals/Ocelot/releases) &rarr; [23.0.0](https://github.com/ThreeMammals/Ocelot/releases/tag/23.0.0) &rarr; **Roadmap** &rarr; **Planning** 
  The quote:
  > Planning of support for **Swagger** and gRPC proto. More details in future release notes.
